Java/BST.java

```diff
@@ -0,0 +1,349 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/** A class to denote a node of the BST.
+ */
+class Node {
```

Comment:
Consider refactoring Node to BSTNode and remove the above comment.

Reply:
Okay will do this change.